docs/examples/Power Flow Example.ipynb

```diff
@@ -348,6 +381,162 @@
     "print(pd.DataFrame(output_data[ComponentType.node]))"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "id": "d129b8de",
+   "metadata": {},
+   "source": [
+    "You can also request the output as a dictionary of attribute and their values columns for each component type."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "id": "95ee7a35",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "List of component types in result dataset\n",
+      "[<ComponentType.line: 'line'>]\n",
+      "List of attribute types in line result\n",
+      "['id', 'p_from']\n",
+      "------line result------\n",
+      "{'id': array([3, 5, 8], dtype=int32), 'p_from': array([17360100.20222374, -3365613.74450156, 13854413.52498137])}\n"
+     ]
+    }
+   ],
+   "source": [
+    "output_data = model.calculate_power_flow(\n",
+    "    symmetric=True,\n",
+    "    error_tolerance=1e-8,\n",
+    "    max_iterations=20,\n",
+    "    calculation_method=CalculationMethod.newton_raphson,\n",
+    "    output_component_types={\n",
+    "        ComponentType.line: [\"id\", \"p_from\"],  # line output columns id and p_from\n",
+    "    },\n",
+    ")\n",
+    "print(\"List of component types in result dataset\")\n",
+    "print(list(output_data))\n",
+    "print(\"List of attribute types in line result\")\n",
+    "print(list(output_data[ComponentType.line]))\n",
+    "print(\"------line result------\")\n",
+    "print(output_data[ComponentType.line])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e564346b",
+   "metadata": {},
+   "source": [
+    "You can also mix output types between components. In this example:\n",
```

Comment:
Maybe we can 
1) Not demo all ways and only explain the functionality of what they would give out.
2) Or maybe only print out the numpy array and only the dict keys.
(I'm saying the whole code block feels repetitive)

Reply:
This is about different usages of output_component_types btw. Must be difficult to identify from notebook diff.

Reply:
what do you think about
```py
print("List of component types in result dataset and associated data types")
print({str(component_type): type(component_data) for component_type, component_data in output_data.items()})
```
with result
```py
List of component types in result dataset and associated data types
{'ComponentType.node': <class 'numpy.ndarray'>, 'ComponentType.line': <class 'dict'>, 'ComponentType.sym_load': <class 'dict'>}
```
resp.

```py
List of component types in result dataset and associated data types
{'ComponentType.node': <class 'dict'>, 'ComponentType.line': <class 'dict'>, 'ComponentType.sym_load': <class 'dict'>, 'ComponentType.source': <class 'dict'>}
```
?

Reply:
or do you prefer

```py
print("List of component types in result dataset")
print(list(output_data.keys()))
print("------node result------")
print(output_data[ComponentType.node].dtype.names)
print("------line result attributes------")
print(list(output_data[ComponentType.line].keys()))
print("------sym_load result attributes------")
print(list(output_data[ComponentType.sym_load].keys()))
```
with output
```py
List of component types in result dataset
[<ComponentType.node: 'node'>, <ComponentType.line: 'line'>, <ComponentType.sym_load: 'sym_load'>]
------node result------
('id', 'energized', 'u_pu', 'u', 'u_angle', 'p', 'q')
------line result attributes------
['id', 'p_from']
------sym_load result attributes------
['id', 'energized', 'p', 'q', 'i', 's', 'pf']
```
resp.
```py
List of component types in result dataset
[<ComponentType.node: 'node'>, <ComponentType.line: 'line'>, <ComponentType.sym_load: 'sym_load'>, <ComponentType.source: 'source'>]
------node result------
['id', 'energized', 'u_pu', 'u', 'u_angle', 'p', 'q']
------line result attributes------
['id', 'energized', 'loading', 'p_from', 'q_from', 'i_from', 's_from', 'p_to', 'q_to', 'i_to', 's_to']
------sym_load result attributes------
['id', 'energized', 'p', 'q', 'i', 's', 'pf']
```
?

Reply:
I guess the first way to demonstrate row / columnar data difference.

Reply:
went with a combination of the two because i do think it's important to show the actual output types